### PR TITLE
New design reader bottom sheet - step 1

### LINKED ISSUE #11017: IA Reader: collapse/uncollapse strategy in the bottom sheet
In the IA reviewed reader filter bottom sheet, we need to better define what to present to the user in cases where followed sites/tags are huge.

A possible solution could be to put in place some collapse/uncollapse strategy in the bottom sheet for Sites list and Tags list.

cc @osullivanchris

### PR DESCRIPTION
**Pull request by develric**
Fixes #11017 and partially implements #10887 

Tried to separate the modifications a bit to not make all in one God PR.

| Phone | image | image |
|---|---|---|
| Tablet | image | image |

## Notes
The feature is active in the `zalpha` flavor.

What's covered in this PR:
1. The subfilter (the one with the settings cog) uses the `Filter` call to action when no filter is applied. When a filter is applied an `x` is presented near the filter name so that the user can remove the filter tapping on it
2. Bottom sheet now has `SITES` and `TAGS` tabs and on the header part presents rounded corners and a central handle
3. The bottom sheet is half expanded when opened both in portrait and landscape mode. 

## To test
The following test cases test the list of items reported above (same numbering applies)

### Test case 1
- Follow some sites and tags if not already
- Check that when no site/tag is selected in the subfilter it presents the `Filter` call to action
- Select a site/tag from the bottom sheet and check the correct label is presented in the subfilter and the posts from that stream are loaded in the reader
- Remove the filter (pressing on the `x` near the filter label) and check that again the `Filter` label is presented

### Test case 2
- Follow some sites and tags if not already
- Open the bottom sheet and check it is tabbed like the images above
- Check that the tabs title shows the number of followed sites and tags
- Smoke test the two lists checking the correct filter is selected and correct posts are loaded

### Test case 3
- Open the bottom sheet (both landscape and portrait) and check it is half height of the screen
- Same if rotating the device while bottom sheet is opened
- The half height is also enforced when the bottom sheed is expanded to full height and then collapsed.

PR submission checklist:
note: not adding accessibility here since there will be a dedicated audit of the feature to close ticket #10962 

- [x] I have considered adding unit tests where possible.
- [x] I have considered adding accessibility improvements for my changes.
- [x] I have considered if this change warrants user-facing release notes and have added them to `RELEASE-NOTES.txt` if necessary.

### CODE REVIEW

**Comment by develric** [comment]
Hi @osullivanchris 😊!

**Drag handle**
I confirm the width is 32dp and height is 5dp, but you are right we are using a slightly darker color:

`<color name="gray_5">#dcdcde</color>`

the one we use for dividers semantically. Here how it appears using material grey 300.

| current | gray 300 |
|---|---|
| image | image |

**Sheet heading "Following"**
It's using this, but strange it looks not correct, I wonder if there is something wrong with my screenshots 🤔.

`<dimen name="text_sz_large">16sp</dimen>`

**'X' to clear filter**
It's strange but it seems 2dp top padding makes the trick also at different dpi devices

<p align="center">
  
</p>

**Comment by osullivanchris** [comment]
> <color name="gray_5">#dcdcde</color> the one we use for dividers semantically. Here how it appears using material grey 300.

oh thats fine then lets stick with gray 5. Thats more up to date and using the correct semantic color system so thats fine. Keep it the way it was.

> It's using this, but strange it looks not correct, I wonder if there is something wrong with my screenshots 🤔.<dimen name="text_sz_large">16sp</dimen>

thats also fine as long as we are using the native text styles. 16sp as expected. I was using a screenshot from my own device so it might just be appearing slightly different depending on the resolution. No change required.

> 'X' to clear filter
It's strange but it seems 2dp top padding makes the trick also at different dpi devices

that fixes the issue for me, thanks! Lets make that change

File: WordPress/src/main/java/org/wordpress/android/ui/reader/ReaderPostListFragment.java [hunk 1]
```diff
@@ -970,6 +977,11 @@ public void onClick(View view) {
             });
 
             mSubFilterTitle = mSubFilterComponent.findViewById(R.id.selected_filter_name);
+
+            mRemoveFilterButton = mSubFilterComponent.findViewById(R.id.remove_filter_button);
+            mRemoveFilterButton.setOnClickListener(v -> {
+                mViewModel.setDefaultSubfilter();
```
[the remark this answers is not in the document]
Comment (develric) [reply WordPress/src/main/java/org/wordpress/android/ui/reader/ReaderPostListFragment.java]: Hi @khaykov , totally agree with this 👍! If it's fine for you I would ask to merge this one and then I will change the View Model naming in the next related [step 2 PR](https://github.com/wordpress-mobile/WordPress-Android/pull/11184) where I will port (merge) the content of this one. (btw Chris already design approved step 2 👍). Let me know wdyt 😊!

[the remark this answers is not in the document]
Comment (khaykov) [reply WordPress/src/main/java/org/wordpress/android/ui/reader/ReaderPostListFragment.java]: Yeah, that's totally fine with me 👍

**Pull request merged**